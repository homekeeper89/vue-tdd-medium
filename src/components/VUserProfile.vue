<template>
  <div class="user-profile">
<<<<<<< HEAD
    <img class="user-profile__avatar" :src="user.avatar_url">
    <div class="user-profile__name">
      {{user.name}}
    </div>
    <div class="user-profile__bio">
      {{user.bio}}
    </div>
=======
    <img class ='user-profile__avatar' :src='user.avatar_url'>
    <div class="user-profile__name">{{user.name}}</div>
    <div class="user-profile__bio">{{user.bio}}</div>
>>>>>>> feature/testing-VUserProfile-3
  </div>
</template>
<script>
export default {
  name: 'UserProfile',
  props:{
    user:{
      type:Object,
      required:true,
      default:()=>({})
    }
  }
}
</script>
