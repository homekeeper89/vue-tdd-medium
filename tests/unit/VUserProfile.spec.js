import { shallowMount} from '@vue/test-utils'
import VUserProfile from '@/components/VUserProfile'
import user from './fixtures/user'

describe('VUserProfile이 제대로 렌더 되는지 확인', ()=>{
  let props
  const build = ()=>{
    const wrapper = shallowMount(VUserProfile, {
      propsData:props
    })
    return {
      wrapper,
      avatar: ()=>wrapper.find('.user-profile__avatar'),
      name:()=>wrapper.find('.user-profile__name'),
      bio:()=>wrapper.find('.user-profile__bio')
    }
  }

  beforeEach(()=>{
    props = {
      user
    }
  })

  it('render main html', ()=>{
    const {wrapper} = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('render user template', ()=>{
    const
  })
})
