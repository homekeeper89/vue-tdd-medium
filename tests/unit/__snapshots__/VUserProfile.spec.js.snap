// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUserProfile render main components 1`] = `[Function]`;

exports[`VUserProfile이 제대로 렌더 되는지 확인 render main html 1`] = `
<div>
  UserProfile
</div>
`;
